Sketch.java

```diff
@@ -1,36 +1,137 @@
 import processing.core.PApplet;
+import processing.core.PImage;
+import java.util.Set;
+import java.util.HashSet;
+
 
 public class Sketch extends PApplet {
 	
-	
+  Snow[] circles;
+  Player player;
+  public static final int SIZE = 400;
+  int boost = 0;
+  Set<Snow> hitSnowBalls = new HashSet<>();
+  
   /**
    * Called once at the beginning of execution, put your size all in this method
    */
   public void settings() {
-	// put your size call here
-    size(400, 400);
+	  // put your size call here
+    size(SIZE, SIZE);
   }
 
   /** 
    * Called once at the beginning of execution.  Add initial set up
    * values here i.e background, stroke, fill etc.
    */
   public void setup() {
-    background(210, 255, 173);
+    // set up five snowballs
+    circles = new Snow[5];
+    for (int i = 0; i < 5; ++i) {
+      circles[i] = new Snow();
+    }
+
+    // set up one player
+    player = new Player();
   }
 
   /**
-   * Called repeatedly, anything drawn to the screen goes here
-   */
+  * Returns true if x1, y1 and x2, y2 are close to each other (i.e. a hit)
+  */
+  public boolean isHit(int x1, int y1, int x2, int y2) {
+    return ((Math.abs(x1 - x2) < 13) &&
+            (Math.abs(y1 - y2) < 13));  
+  }
+
+  /**
+  * This function draws the screen.
+  */
   public void draw() {
-	  
-	// sample code, delete this stuff
-    stroke(128);
-    line(150, 25, 270, 350);  
+    background(50);
+
+    fill(255, 255, 255);
+    for (int i = 0; i < 5; ++i) {
+      if (!circles[i].ballHideStatus) {
+        // reduce lives when hit by snow
+        if (isHit(circles[i].x, circles[i].y, player.x, player.y)) {
```

Comment:
lives are not lost on collision

Reply:
lives are lost on collision.  In code line 78 to 91, as more and more hits are registered, the number of squares decrease.

Reply:
lives are lost on collision. In code line 78 to 91, as more and more hits are registered, the number of squares decrease.